zenodo/modules/records/utils.py

```diff
@@ -98,3 +105,37 @@ def is_valid_openaire_type(resource_type, communities):
     is_defined = any(t['id'] == oa_subtype for t in subtypes)
     comms_match = len(set(communities) & set(defined_comms))
     return is_defined and comms_match
+
+
+def find_registered_doi_pids(from_date, until_date, prefixes):
+    """Find all local DOI's which are in the REGISTERED state."""
+    query = db.session.query(PersistentIdentifier).filter(
+        PersistentIdentifier.pid_type == 'doi',
+        PersistentIdentifier.status == PIDStatus.REGISTERED,
+        PersistentIdentifier.updated.between(from_date, until_date)
+    )
+
+    query.filter(or_(PersistentIdentifier.pid_value.like(prefix + '%') for prefix in prefixes))
+
+    query.order_by(PersistentIdentifier.updated)
+
+    return query
+
+
+def xsd41():
```

Comment:
It's nice to have this here, though we should consider at some point moving it to `inveniosoftware/datacite` utils. WDYT @lnielsen?

Reply:
Let's remember to make an issue after merging this PR for moving it to `inveniosoftware/datacite`